parsl/executors/workqueue/executor.py

```diff
@@ -402,6 +407,7 @@ def __init__(self,
                  port=WORK_QUEUE_DEFAULT_PORT,
                  env=None,
                  shared_fs=False,
+                 source=False,
```

Comment:
I think this default means that the new code path has not been tested in CI?

Reply:
I tried the workqueue pytest stuff from `.travis.yml` on my laptop with source=True and I get this error pretty much straight away:

```
E           _pickle.PicklingError: Can't pickle <function echo_to_file at 0x7f387eb802f0>: it's not the same object as parsl.tests.test_bash_apps.test_apptimeout.echo_to_file
```

Reply:
Where was the error traced back to? The goal of this PR is to avoid pickling a function object altogether, instead obtaining the source code and transferring that as a string to the worker.

Reply:
Also, I can update the code to change source=True for the build

Reply:
Try in CI - I won't be able to look at this for a few more hours